pages/api/check-status.ts

```diff
@@ -81,24 +81,36 @@ export default async function handler(
   req: NextApiRequest,
   res: NextApiResponse<CheckStatusReponse | string>
 ) {
-  if (req.method !== 'POST') {
-    return res.status(405).send(`Invalid request method ${req.method}`)
+  if (req.method !== 'GET') {
+    res.status(405).send(`Invalid request method ${req.method}`)
+    return
   }
 
-  const body = JSON.parse(req.body) as CheckStatusRequestBody
+  const { searchParams } = new URL(req.url ?? '', `http://${req.headers.host}`)
+  const checkStatusRequest: CheckStatusRequest = {
+    birthDate: searchParams.get('birthDate') ?? '',
+    esrf: searchParams.get('esrf') ?? '',
+    givenName: searchParams.get('givenName') ?? '',
+    surname: searchParams.get('surname') ?? '',
+  }
 
   try {
     const passportStatusApiBaseUri = process.env.PASSPORT_STATUS_API_BASE_URI
     const response = passportStatusApiBaseUri
-      ? await fetchPassportStatusAPI(passportStatusApiBaseUri, body)
-      : await fetchPassportStatusMOCK(body)
-    return res.status(200).json(response)
+      ? await fetchPassportStatusAPI(
+          passportStatusApiBaseUri,
+          checkStatusRequest
+        )
+      : await fetchPassportStatusMOCK(checkStatusRequest)
+    res.status(200).json(response)
   } catch (error) {
-    logger.error(error)
     if ((error as Error).constructor.name === 'ApiError') {
       const apiError = error as ApiError
-      return res.status(apiError.statusCode).send(apiError.message)
+      res.status(apiError.statusCode).send(apiError.message)
+      return
     }
+
+    logger.error(error)
```

Comment:
We want to log all the errors, not just the ones that make it this far; this should be moved to the top of the catch.

Reply:
Will do, thanks.

Reply:
I think this api handler could be revisit to remove some throwing of ApiError. Will take a note.